tests/integ/test_function.py

```diff
@@ -523,6 +524,19 @@ def test_basic_string_operations(session):
     assert "'REVERSE' expected Column or str, got: <class 'list'>" in str(ex_info)
 
 
+def test_substring_index(session):
+    df = session.create_dataframe(["a.b.c.d"], ["s"])
```

Comment:
what will happen if it's an empty string or None?

Reply:
That's a good question, for empty, it will be empty, and same for none, it remains none. Let me add test for it